extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchHolder.java

```diff
@@ -295,6 +296,14 @@ public boolean equals(Object o)
     if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    return this.getSketch().equals(((SketchHolder) o).getSketch());
+    // Can't use Sketch's equals method because it is not implemented.
+    // return this.getSketch().equals(((SketchHolder) o).getSketch());
+    return Arrays.equals(this.getSketch().toByteArray(), ((SketchHolder) o).getSketch().toByteArray());
```

Comment:
I think it would be fine to keep these `equals` and `hashCode` methods just based on `Sketch`. So do equals as:

```
return this.getSketch().equals(((SketchHolder) o).getSketch());
```

And add hashCode as:

```
return this.getSketch().hashCode();
```

The reason I think this is ok is that at least they're consistent. They're reference-based, not value-based, but I think that's ok for now as long as they're consistent. I think we don't have a need for them to be value-based at this time.

And my feeling is that if we ever do need value-based equals/hashCode methods for SketchHolder, it should be done as part of Sketch.

Reply:
I have added some unit test for post aggregation serde in SketchAggregationTest class. *testSketchEstimatePostAggregatorSerde
*testSketchSetPostAggregatorSerde

Since Sketch class doesn't have equals implementation, the above UT will fail.
Hence I updated SketchHolder's equals and hashcode method and not to use Sketch's equals or hashcode.

If I implement the above changes to use Sketch's equals and hashcode then I need to remove serde unit test for SketchConstantPostAggregator.

Please check the unit test class and suggest me how to proceed.

Reply:
Ah I see the problem you are facing. Okay, in that case I would say go ahead with how you did it - base the equals and hashCode off the toByteArray form. Just please include a comment that it's done this way because Sketch impls don't tend to have value based hashCodes and equals, yet we want the SketchHolder to have one.

Reply:
Updated the comments please check once.

Thanks for your time and review comments Gian 👍. Hope i didn't trouble you much.  :)